04-backend-api.md

```diff
@@ -0,0 +1,191 @@
+    Properties:
+      Auth:
+        Authorizers:
+          WildRydes:
+            UserPoolArn: !GetAtt UserPool.Arn
+      Name: !Sub
+        - ${ResourceName} From Stack ${StackTagName} Environment ${EnvironmentTagName}
+        - ResourceName: Api
+      StageName: !Ref EnvironmentAPIGatewayStageName
+      DefinitionBody:
+        swagger: '2.0'
+        info: {}
+        paths:
+          /ride:
+            post:
+              security:
+                - WildRydes: []
+              x-amazon-apigateway-integration:
+                httpMethod: POST
+                type: aws_proxy
+                uri: !Sub arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${RequestUnicorn.Arn}/invocations
+              responses: {}
+      EndpointConfiguration: REGIONAL
+      Cors:
+        AllowOrigin: '''*'''
+        AllowHeaders: '''Authorization,Content-Type'''
+
+```
+
+Finally, commit your changes by clicking the **Commit** button and in the popup window clicking **Commit and Push**.
+
+![Commit And Push](./images/04-commit-and-push.png)
+
+
+
+### 7. Update *RequestUnicorn* Function.
+
+Update the *RequestUnicorn* Function code so it is functional. The code accepts requests and performs the following actions:
+
+* Validates the User Pool Client authentication token (provided via the *Authorization* header from API Gateway) and retrieves the username from it
+* Retrieves the Unicorn Stables™ api key from AWS Secrets Manager and caches the value for subsequent requests
+* Makes a request to the Unicorn Stables™ endpoint to rent a unicorn
+* Records the ride to the *Rides* DynamoDB table
+* Returns with the response to the frontend website request
+
+Start by updating your local *stackery-wild-rydes* GitHub clone (created in module 1) so it has all the changes you've made over the past few workshop modules.
+
+```
+$ cd stackery-wild-rydes     # if not already in the directory.
+$ git fetch origin
+$ git rebase origin/master
+```
+
+Now copy [src/requestUnicorn/index.js](src/requestUnicorn/index.js) from the workshop directory (also created in module 1) into your project directory.
+
+*Note: Make sure you are still in the stackery-wild-rydes project directory when you run the following command.*
+```
+$ cp ../wild-rydes-workshop/src/requestUnicorn/index.js ./src/requestUnicorn/index.js
+```
+
+```
+$ git add src/requestUnicorn/index.js
+$ git commit -a -m "add requestUnicorn"
+$ git push -v
+```
+
+
+
+### 8. Deploy updated Wild Rydes
+
+You'll now deploy the updated *stackery-wild-rydes* stack. Click **Deploy** in the left sidebar to enter the Deploy view. Next click **Prepare new deployment** for the **development** environment. For the **branch or SHA** value enter `master` and then click **Prepare Deployment**. Once the preparation completes (this should take about 20 seconds), click the **Deploy** button to open AWS CloudFormation. Then click the **Execute** button in the CloudFormation Console.
+
```

Comment:
This would be a great place to add something like:

> Pro Tip: You can also use the Stackery CLI to deploy the stack in one step from the command line. You can find instructions [here](https://docs.stackery.io/docs/api/cli/stackery_deploy/).

Reply:
reolved in 3b9b0c8